flatgfa/src/packedseq.rs

```diff
@@ -45,60 +56,98 @@ impl From<Nucleotide> for char {
 /// A compressed vector-like structure for storing nucleotide sequences
 ///     - Two base pairs are stored per byte
 ///
-pub struct PackedVec {
-    /// A vector that stores a compressed encoding of this PackedVec's sequence
+pub struct PackedSeqStore {
+    /// A vector that stores a compressed encoding of this PackedSeqStore's sequence
     data: Vec<u8>,
 
     /// True if the final base pair in the sequence is stored at a
     ///                   high nibble
     high_nibble_end: bool,
 }
 
-impl PackedVec {
-    /// Creates a new empty PackedVec
-    pub fn new() -> Self {
-        PackedVec {
-            data: Vec::new(),
-            high_nibble_end: true,
-        }
+pub struct PackedSeqView<'a> {
+    data: &'a [u8],
+
+    /// True if the final base pair in the sequence is stored at a
+    ///                   high nibble
+    high_nibble_end: bool,
+}
+
+#[derive(FromBytes, FromZeroes, AsBytes, Debug)]
+#[repr(packed)]
+pub struct PackedToc {
+    magic: u8,
```

Comment:
I know this is a silly thing, but you probably want the [magic number](https://en.wikipedia.org/wiki/File_format#Magic_number) to be an entire word, i.e., a `u64`. Here are two reasons:
* There are just too many collisions in the space of a `u8`! That is, a `u64` that you pick is likely to be unique; a `u8` that you pick is likely to be the same as someone else's `u8` magic number.
* Using a single byte at the start of the file will make the rest of the data unaligned. If the magic number is a `u64`, then the next field will start on a word-aligned boundary.